[ews] Dynamically fetch approved user list for Apple Internal builds

aj062 opened the pull request:
#### 7d93113afc723c94517fe8f3b8f3d8fea7527629
<pre>
[ews] Dynamically fetch approved user list for Apple Internal builds
<a href="https://bugs.webkit.org/show_bug.cgi?id=296625">https://bugs.webkit.org/show_bug.cgi?id=296625</a>
<a href="https://rdar.apple.com/157017667">rdar://157017667</a>

Reviewed by Brianna Fan.

* Tools/CISupport/ews-app/ews/common/github.py:
(GitHub):
(GitHub.fetch_data_from_url_with_authentication_github):
(GitHubEWS):
(GitHubEWS.update_approved_user_list_for_apple_internal_builds):
* Tools/CISupport/ews-app/ews/fetcher.py:

Canonical link: <a href="https://commits.webkit.org/298187@main">https://commits.webkit.org/298187@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/bd0caba4c14d12f6ff7a66505ac8efbe721e156e

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows | Apple Internal |
| ----- | ---------------------- | ------- |  ----- |  --------- | ------ |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/114048 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/33800 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/24258 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/120218 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/65029 "Built successfully") | [✅ 🛠 ios-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/92699e7b-e2c4-41dd-8b79-0fc7c00d5b52/34070654-7f74-41d1-8727-67a5cf581fc5) 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/115937 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/34428 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/42361 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/86679 "Passed tests") | [❌ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/41787 "") | [✅ 🛠 mac-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/92699e7b-e2c4-41dd-8b79-0fc7c00d5b52/0d4568e9-dbf7-4f20-ba6a-76428cd9e661) 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/116996 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/27408 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/102431 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/67062 "Passed tests") | | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/26593 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/20560 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/63924 "Built successfully") | | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/96776 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/20677 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/123445 "Built successfully") | | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/41081 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/30605 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/95514 "Passed tests") | | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/41459 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/98641 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/95296 "Passed tests") | | 
| | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/40426 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/18238 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/37243 "Built successfully") | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/18349 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/40956 "Built successfully") | | | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/40583 "Built successfully") | | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/43882 "Built successfully") | | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/42337 "Built successfully") | | | | 
<!--EWS-Status-Bubble-End-->

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/2ed2b1f7420eb12c47f2131246b3f557aa7f494e. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/48655

briannafan commented on Tools/CISupport/ews-app/ews/common/github.py:
```diff
@@ -220,11 +221,32 @@ class GitHubEWS(GitHub):
+            url = f'{URL_FOR_TRUSTED_GITHUB_LIST}?per_page={PER_PAGE_LIMIT}&page={page}'
+            content = cls.fetch_data_from_url_with_authentication_github(url)
+            if not content:
+                _log.error(f'ERROR: Unable to fetch list of trusted users, this might impact Internal EWS.')
```
If this happens, is there a step that will propagate up an actionable error message (like contact admin for example)?

briannafan commented on Tools/CISupport/ews-app/ews/common/github.py:
```diff
@@ -220,11 +221,32 @@ class GitHubEWS(GitHub):
+            members.extend(user['login'] for user in response_content)
+            if len(response_content) < PER_PAGE_LIMIT:
+                break
+            page += 1
```
Is this necessary since the loop is over `page in range(1, NUM_PAGE_LIMIT + 1)`?

aj062 replied:
Good catch. This is basically a dead line of code. 
We are doing the same thing in ews-build in https://github.com/WebKit/WebKit/blob/main/Tools/CISupport/ews-build/steps.py#L336

Removed in updated PR.

briannafan commented on Tools/CISupport/ews-app/ews/common/github.py:
```diff
@@ -220,11 +221,32 @@ class GitHubEWS(GitHub):
+
+        _log.info(f'{len(members)} Members: \n{members}')
+        if members:
+            GitHubEWS.approved_user_list_for_apple_internal_builds = members
```
What happens if members is empty?

briannafan reviewed (approved):
Thanks! LGTM

aj062 committed:
[ews] Dynamically fetch approved user list for Apple Internal builds
https://bugs.webkit.org/show_bug.cgi?id=296625
rdar://157017667

Reviewed by Brianna Fan.

* Tools/CISupport/ews-app/ews/common/github.py:
(GitHub):
(GitHub.fetch_data_from_url_with_authentication_github):
(GitHubEWS):
(GitHubEWS.update_approved_user_list_for_apple_internal_builds):
* Tools/CISupport/ews-app/ews/fetcher.py:

Canonical link: https://commits.webkit.org/298187@main
